### ISSUE TITLE
node-s3-cli with Digital Ocean Spaces

### ISSUE THREAD

**Issue by gsiotas** [andrewrk/node-s3-cli#39]
After searching through and following this [comment](https://github.com/andrewrk/node-s3-cli/issues/37#issue-287499846), I managed to make the node-s3-cli work with [Digital Ocean Spaces](https://www.digitalocean.com/community/tutorials/an-introduction-to-digitalocean-spaces) (DO Spaces).
In the config file "~/.s3cfg" just put the default configuration:
```
[default]
access_key = foo
secret_key = bar
```

Of course you have to change the foo and the bar according to the data you receive from [DO API page](https://cloud.digitalocean.com/settings/api/tokens).

Then on the file cli.cfg, usually found at:
> /usr/local/lib/node_modules/s3-cli

you change the following:
```
  client = s3.createClient({
    s3Options: {
      accessKeyId: accessKeyId,
      secretAccessKey: secretAccessKey,
      sslEnabled: !args.insecure,
      region: args.region,
endpoint: 'ams3.digitaloceanspaces.com',
    },
```

Basically you enter the endpoint **endpoint: 'ams3.digitaloceanspaces.com',** change according to the location of your space.

If I won't hardcore the endpoint I get this error message:

> Error: The AWS Access Key Id you provided does not exist in our records.

Thank you!

**Comment by thecatontheflat** [comment]
Thank you for figuring this, it helped!

**Comment by KieronWiltshire** [comment]
I'll fork the project and submit a pull request so that maybe this will be included in the next release. Thanks for figuring this out!
